### PR TITLE
[CARBONDATA-3803] Mark CarbonSession as deprecated since 2.0

### PR DESCRIPTION
**Pull request by QiangCai**
### Why is this PR needed?
 Better to use CarbonExtensions instead of CarbonSession in version 2.0.
 
 ### What changes were proposed in this PR?
mark CarbonSession as deprecated since version 2.0.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No, no need

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1247/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2965/

**Comment by kunal642** [comment]
Please print a log in CarbonSession to indicate to the user that this is deprecated

File: integration/spark/src/main/scala/org/apache/spark/sql/CarbonSession.scala [hunk 1]
```diff
@@ -40,12 +40,18 @@ import org.apache.carbondata.streaming.CarbonStreamingQueryListener
  * Session implementation for {org.apache.spark.sql.SparkSession}
  * Implemented this class only to use our own SQL DDL commands.
  * User needs to use {CarbonSession.getOrCreateCarbon} to create Carbon session.
+ *
+ * @deprecated Since 2.0, only use for backward compatibility,
+ *             please switch to use {@link CarbonExtensions}.
  */
+@Deprecated
 class CarbonSession(@transient val sc: SparkContext,
```
Comment (ajantha-bhat) [inline integration/spark/src/main/scala/org/apache/spark/sql/CarbonSession.scala:48]: "**stored by**" syntax also we need to deprecate and update document for the same?

Comment (QiangCai) [reply integration/spark/src/main/scala/org/apache/spark/sql/CarbonSession.scala:48]: the document already contains it

Comment (QiangCai) [reply integration/spark/src/main/scala/org/apache/spark/sql/CarbonSession.scala:48]: please check https://github.com/apache/carbondata/blob/master/docs/quick-start-guide.md,
it lists all supported syntax by CarbonSession and CarbonExtensions.

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1256/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2974/

**Comment by QiangCai** [comment]
> Please print a warn log in CarbonSession to indicate to the user that this is deprecated

done

**Comment by ajantha-bhat** [comment]
LGTM

**Pull request closed without merging**